### ISSUE TITLE
Creating arrays from slices

### ISSUE THREAD

**Issue by rennis250** [arrayfire/arrayfire-rust#121]
It seems you cannot create arrays that have 0 elements in the 3rd and 4th dimensions? Using ```moddims()``` doesn't change this. For example, the following snippet of code fails:

```rust
extern crate arrayfire as af;

use af::*;

fn main() {
    set_backend(Backend::CPU);

    let a = Array::new(&vec![0f32; 100], Dim4::new(&[10, 10, 0, 0])); // incorrect size error
}
```

Best,
Rob

**Comment by pavanky** [comment]
@rennis250 are you trying to create [10, 10, 1, 1] or [10, 10, 0, 0] ? Arrayfire does not handle empty arrays gracefully at all so the error is kind of expected.

**Comment by rennis250** [comment]
I would ultimately like something of size (10, 10, 0, 0), since I suspect my problems with matrix multiplication in #122 is related.

**Comment by rennis250** [comment]
Hmmm... seems I mainly solved the problem in #122.
